Line2D[] lines
bool isPathFree
int32[] indices_left
int32[] indices_right
int32[] indices_horizontal
int32[] indices_vertical
int32[] indices_obstacle 
int32[] indices_uninteresting